File: src/types/project.ts
import type { AssetIndex } from "./asset";
import type { LevelSummary } from "./scene";

export interface AppDirectories {
  documentsRoot: string;
  projectsDir: string;
  templatesDir: string;
  backupsDir: string;
  localRoot: string;
  cacheDir: string;
  logsDir: string;
  shaderCacheDir: string;
  tempDir: string;
  roamingRoot: string;
  settingsPath: string;
}

export interface ProjectManifest {
  projectId: string;
  projectName: string;
  description: string;
  engineVersion: string;
  createdAt: string;
  lastOpenedAt: string;
  rootPath: string;
  defaultScene: string | null;
  projectSettingsPath: string;
  contentRoot: string;
  assetIndexPath: string;
}

export interface ProjectSummary {
  projectId: string;
  projectName: string;
  rootPath: string;
  manifestPath: string;
  lastOpenedAt: string;
  pinned: boolean;
}

export interface CreateProjectRequest {
  projectName: string;
  location: string;
  description: string;
  template: string;
  renderBackend: string;
  targetPlatform: string;
  starterContent: boolean;
  sourceControlIgnore: boolean;
  createDefaultScene: boolean;
}

export interface OpenProjectResponse {
  manifest: ProjectManifest;
  levels: LevelSummary[];
  assetIndex: AssetIndex;
}

export interface ProjectValidation {
  valid: boolean;
  manifestPath: string | null;
  message: string;
}

File: src-tauri/src/models/project.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ProjectManifest {
    pub project_id: String,
    pub project_name: String,
    pub description: String,
    pub engine_version: String,
    pub created_at: String,
    pub last_opened_at: String,
    pub root_path: String,
    pub default_scene: Option<String>,
    pub project_settings_path: String,
    pub content_root: String,
    pub asset_index_path: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ProjectSummary {
    pub project_id: String,
    pub project_name: String,
    pub root_path: String,
    pub manifest_path: String,
    pub last_opened_at: String,
    pub pinned: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateProjectRequest {
    pub project_name: String,
    pub location: String,
    pub description: String,
    pub template: String,
    pub render_backend: String,
    pub target_platform: String,
    pub starter_content: bool,
    pub source_control_ignore: bool,
    pub create_default_scene: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct OpenProjectResponse {
    pub manifest: ProjectManifest,
    pub levels: Vec<crate::models::scene::LevelSummary>,
    pub asset_index: crate::models::asset::AssetIndex,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ProjectValidation {
    pub valid: bool,
    pub manifest_path: Option<String>,
    pub message: String,
}

